framework/src/Volo.Abp.EventBus.Azure/Volo/Abp/EventBus/Azure/AzureDistributedEventBus.cs

```diff
@@ -108,9 +108,14 @@ public async override Task PublishManyFromOutboxAsync(IEnumerable<OutgoingEventI
 
         foreach (var outgoingEvent in outgoingEventArray)
         {
-            if (!messageBatch.TryAddMessage(new ServiceBusMessage(outgoingEvent.EventData) { Subject = outgoingEvent.EventName }))
+            var message = new ServiceBusMessage(outgoingEvent.EventData) {
```

Comment:
```csharp
var message = new ServiceBusMessage(outgoingEvent.EventData) 
{
    Subject = outgoingEvent.EventName
};

if (message.MessageId.IsNullOrWhiteSpace())
{
    message.MessageId = outgoingEvent.Id.ToString();
}
```

Reply:
Should we change the code in `PublishAsync` too?

Replace
```CSharp
message.MessageId = (eventId ?? GuidGenerator.Create()).ToString("N");
```
with
```CSharp
message.MessageId = outgoingEvent.Id.ToString();
```

Reply:
Should not change the code, `eventId` is a nullable parameter, 

https://github.com/abpframework/abp/blob/48ed43da234e99e97c6f617175b99d4f445d6ba1/framework/src/Volo.Abp.EventBus.Azure/Volo/Abp/EventBus/Azure/AzureDistributedEventBus.cs#L228